packages/core/src/UserState.ts

```diff
@@ -173,22 +170,20 @@ export default class UserState extends Synchronizer {
 
     /**
      * Get the reputation object from a given attester
-     * @param _attesterId The attester ID that the user queries
      * @param toEpoch The latest epoch that the reputation is accumulated
      * @returns The reputation object
      */
     public getRepByAttester = async (
-        _attesterId?: bigint | string,
-        toEpoch?: number
+        _toEpoch?: number
     ): Promise<{ posRep; negRep; graffiti; timestamp }> => {
         let posRep = BigInt(0)
         let negRep = BigInt(0)
         let graffiti = BigInt(0)
         let timestamp = BigInt(0)
-        const attesterId = _attesterId ?? this.attesterId
         const orClauses = [] as any[]
-        const latestTransitionedEpoch = await this.latestTransitionedEpoch()
-        for (let x = 0; x < (toEpoch ?? latestTransitionedEpoch); x++) {
+        const attesterId = this.attesterId
+        const toEpoch = _toEpoch ?? this.calcCurrentEpoch()
+        for (let x = 0; x <= toEpoch; x++) {
```

Comment:
ah actually I prefer
```ts
const toEpoch = _toEpoch ?? this.calcCurrentEpoch() - 1
for (let x = 0; x <= toEpoch; x++) {
```

since it is what user already receive and it won't confuse users when they prove reputation

Reply:
Ah no the user can only prove up to `latestTransitionedEpoch - 1`, not `currentEpoch - 1`. Maybe it's better to add a helper function like `getProvableRep()`.

Reply:
I think user can prove up to `latestTransitionedEpoch` and `latestTransitionedEpoch` always less than `currentEpoch`
so it can be
```ts
const toEpoch = _toEpoch ?? await this.latestTransitionedEpoch()
for (let x = 0; x <= toEpoch; x++) {
```

since `latestTransitionedEpoch()` finds the latest nullifier from `currentEpoch`
but the nullifier `hash(id, attester, currentEpoch)` cannot be existed until next epoch
because the circuit computes nullifiers with `fromEpoch`

https://github.com/Unirep/Unirep/blob/a671ac3dc987ba69c34c6c22bd87294addb49ee0/packages/circuits/circuits/userStateTransition.circom#L152

and the `fromEpoch` cannot equal to `currentEpoch`

Reply:
so my point is that
- If it uses `latestTransitionedEpoch` here, the document cannot be `currentEpoch - 1` (Though it is changed. And whatever we use as default `toEpoch` here the document should be clear)
- And the `currentEpoch` rep had better not to be included because it is not provable
- `latestTransitionedEpoch` < `currentEpoch`, so I think it can be the default `toEpoch`

Reply:
Right that's how it originally was. It's not useful to the developer though because they might want to display the most up to date balance. That's why i suggested adding a different function that is `getProvableRep`.

Reply:
@vivianjeng planning to leave this as is. We can add a `getProvableRep` function in a separate pr.

Reply:
I think we change the `<` to `<=` is useful enough for developer to use `getRepByAttester(currentEpoch)`
and then we don't need another function `getProvableRep`
I think `currentEpoch` as default value might be dangerous(!?) and users might not notice there is another function `getProvableRep`

Reply:
That would require the dev to do `getRepByAttester(currentEpoch + 1)` to get rep to the current epoch. This is even less clear than having a separate function.

> I think currentEpoch as default value might be dangerous(!?)

I think it's unfair to prospective developers to assume this. If a developer uses a bad value for `posRep`/`negRep` the proof will be invalid, or the tree inclusion will be invalid.

Reply:
> That would require the dev to do `getRepByAttester(currentEpoch + 1)` to get rep to the current epoch. This is even less clear than having a separate function.

No we just need `getRepByAttester(currentEpoch)` we can get rep in current epoch
if we change `<` to `<=` (like the current commitment)
https://github.com/Unirep/Unirep/blob/b217d74dd5d56a32a1d1a09e72b99fc2cc84a09d/packages/core/src/UserState.ts#L186

Reply:
oh right i misread and thought you wanted to change `<=` to `<`.

Reply:
so the current implementation is fine then?

Reply:
I prefer
```ts
const toEpoch = _toEpoch ?? await this.latestTransitionedEpoch()
for (let x = 0; x <= toEpoch; x++) {
```

and use `getRepByAttester(currentEpoch)` to get the updated rep

Reply:
but what is the `getRepByAttester(currentEpoch)` used for?
since it can neither be provable nor be spent (in the current version)

Reply:
It shows the latest user reputation balance. It's an important signal about activity in the system.

Reply:
another point that not using `currentEpoch` as default value is that
getting the latest rep by `getRepByAttester(currentEpoch)` is trivial but getting provable rep by `getRepByAttester(latestTranstionedEpoch)` or `getRepByAttester(currentEpoch-1)` is not trivial if user is not really familiar with the protocol